lib/courses.ts

```diff
@@ -20,15 +20,33 @@ export function getCourse(courseNo: string) {
 }
 
 // Returns all prerequisites of a given course.
-
+// Return two arrays - first array is prerequisites, second array is corequisites
 export function getCoursePrerequisites(courseNo: string) {
+  const prereqs : ICourse[][] = [[], []];
   const courses: ICourse[] = JSON.parse(
     fs.readFileSync(path.join(process.cwd(), '/data/courses.json'), 'utf-8')
   );
   const course = courses.find((course) => course.course_no == courseNo);
-  if (!course?.pre) return [[], []];
-  const prereqCourse = getCourse(course.pre);
-  if (!prereqCourse) return [];
 
-  return [prereqCourse];
+  // Empty prerequisites/corequisites
+  if (!course?.pre) return prereqs;
+
+  const str = course?.pre;
+  let i = 0;
+  // Get prerequisites
+  while (str[i] != "/") {
+    const prereqCourse = getCourse(str.substring(i, i + 6));
```

Comment:
Just to confirm, can we guarantee that every course has 6 characters?

Reply:
This is the only thing that I'm a little bit worried about, but other than that the code looks good

Reply:
Assuming no uncaught mistakes in courses.json then yes (two-digit course numbers still start with a 0 - for example DAN010); if there are mistakes than I think we can fix those when someone finds them

Reply:
Alright, sounds good